src/simplewallet/simplewallet.cpp

```diff
@@ -3214,6 +3215,40 @@ bool simple_wallet::apropos(const std::vector<std::string> &args)
   return true;
 }
 
+bool simple_wallet::scan_tx(const std::vector<std::string> &args)
+{
+  if (args.empty())
+  {
+    PRINT_USAGE(USAGE_SCAN_TX);
+    return true;
+  }
+
+  // Dedup args and make sure they are hashes
+  std::set<string> txids;
+  for (const auto &s : args) {
+    if (!string_tools::validate_hex(64, s)) {
+      fail_msg_writer() << tr("Invalid txid specified");
+      return true;
+    }
+    txids.insert(s);
+  }
+
+  message_writer() << tr("This operation may reveal the txids to the remote node and affect your privacy in case of using an untrusted node");
+  if (!command_line::is_yes(input_line("Do you want to continue?", true))) {
+    message_writer() << tr("You have canceled the operation");
+    return true;
+  }
+
+  m_in_manual_refresh.store(true);
+  try {
+    m_wallet->scan_tx(txids);
```

Comment:
txids is a set, so sorted by txid. This will mess up spend detection. You want to order them chronologically (either by letting hte user give them in order, or sorting them in wallet2, but the latter is not done).

Reply:
Oh, i forgot about chronological ordering.

@hMihaiDavid sorry, using the vector was/is okay.

Reply:
My bad, I was not aware of this restriction. I propose to follow the second option by @moneromooo-monero. Ask the daemon for txs with txids in the order returned by the deduped set (ascending lexicographical ordering), order them in chronological order by the timestamp returned by daemon and process them in that order. If you think this is necessary I will let the function receive a vector instead of a set and warn the user to give the txids in chron order.

Reply:
Correction: unnecessary